data_collection/web/guides.rst

```diff
@@ -386,31 +429,51 @@ JavaScript Tracking Client distinguishes three parts of the content structure: `
+For more details visit the :ref:`Content tracking<jtc-api-content-tracking>` section of the JavaScript Tracking Client API documentation.
 
-For more information visit the :ref:`Content tracking<jtc-api-content-tracking>` section of the JavaScript Tracking Client API documentation.
+.. note::
 
-**But how JavaScript Tracking Client will know what blocks you would like to track?**
-There are two ways of marking the blocks, you should either use a ``piwikTrackContent`` CSS class or a special html attribute ``data-track-content`` on them.
-Same technique is used for pointing out the content piece (``piwikContentPiece`` CSS class or ``data-content-piece`` attribute) and the content target (``piwikContentTarget`` CSS class or ``data-content-target`` attribute).
+    Automatic content tracking can be enabled in Tag Manager, as shown in
+    `Set up content tracking <https://help.piwik.pro/support/questions/set-up-content-tracking/>`_
+    article.
 
-Although JavaScript Tracking Client has the ability of auto-detection for name, piece and target metrics, we still recommend providing those values manually as was described in the previous paragraph. If you don't then JavaScript Tracking Client will try to fill them as follows:
+**But how does JavaScript Tracking Client know what blocks you would like to track?**
 
-* it will read block ``title`` attribute as for the Content name
-* it will read piece from the ``src`` attribute of an image
-* it will read target from the ``href`` attribute of an anchor wrapping the image
+There are two ways of marking HTML elements as content blocks: you must either add a special attribute ``data-track-content`` or class ``piwikTrackContent``. Example:
 
-As you can imagine this may produce inconsistent results, providing those values manually seems like a more desired approach.
+.. code-block:: html
+    :linenos:
+
+    <a href="http://example.com/image/abc.png" title="abc" data-track-content>
+        first content block
+    </a>
+    <a href="http://example.com/image/def.png" title="def" class="piwikTrackContent">
+        second content block
+    </a>
+
+Content properties will be taken from HTML attributes of the content block element or any of its descendants:
+
+* name comes from ``data-content-name`` attribute
+* piece comes from ``data-content-piece`` attribute
+* target comes from ``data-content-target`` attribute
+
+If any of these attributes is missing, JavaScript Tracking Client will try extracting the value from other sources, using the following logic:
+
+* piece will be taken from ``src`` attribute of an element with ``piwikContentPiece`` class or block element
+* target will be taken from ``href`` attribute of an element with ``piwikContentTarget`` class, block element or piece element
+* name will be a copy of piece value if present, or taken from ``title`` attribute of block element, piece element or target element
```

Comment:
```suggestion
* name will try to use piece value if present, otherwise it'll be taken from ``title`` attribute of block element, piece element or target element
```